docs/products/opensearch/howto/custom-repositories.md

```diff
@@ -0,0 +1,264 @@
+      },
+      "type": "azure"
+    }
+  ]
+}
+```
+
+## Manage snapshots in custom repositories
+
+After setting up a custom repository, you can use the Aiven API to create, list,
+retrieve, and delete snapshots.
+
+### Create a snapshot
+
+Create a snapshot in a custom repository.
+
+```sh
+curl -s -X POST \
+--url "https://api.aiven.io/v1/project/{project_name}/service/{service_name}/opensearch/_snapshot/aws-repo/first-snapshot" \
+--header "Authorization: Bearer $TOKEN" \
+--header "Content-Type: application/json" \
+-d '{"indices": "test*", "include_global_state": false}'
+```
+
+Example response:
+
+```json
+{
+  "accepted": true
+}
+```
+
+### List snapshots in progress
+
+Retrieve snapshots that are still being created in a repository.
+
+```sh
+curl -s --url "https://api.aiven.io/v1/project/{project_name}/service/{service_name}/opensearch/_snapshot/aws-repo/_status" \
+--header "Authorization: Bearer $TOKEN"
+--header "Content-Type: application/json"
+```
+
+Example response:
+
+```json
+{
+  "snapshots": [
+    {
+      "repository": "aws-repo",
+      "snapshot": "second-snapshot",
+      "state": "SUCCESS",
+      "shards_stats": {
+        "done": 1,
+        "failed": 0,
+        "total": 1
+      },
+      "uuid": "osmCbdF-RMyyUKpWD-4bJA"
+    }
+  ]
+}
+```
+
+### List all snapshots in a repository
+
+Retrieve all snapshots, including completed and failed ones.
+
+```sh
+curl -s --url "https://api.aiven.io/v1/project/{project_name}/service/{service_name}/opensearch/_snapshot/aws-repo/_all" \
+--header "Authorization: Bearer $TOKEN"
+--header "Content-Type: application/json"
+```
+
+Example response:
+
+```json
+{
+  "snapshots": [
+    {
+      "snapshot": "first-snapshot",
+      "state": "SUCCESS",
+      "indices": ["test"],
+      "uuid": "7cdWedW7RC6FMSktlZTCDw"
+    },
+    {
+      "snapshot": "second-snapshot",
+      "state": "SUCCESS",
+      "indices": ["test"],
+      "uuid": "osmCbdF-RMyyUKpWD-4bJA"
+    }
+  ]
+}
+```
+
+### Retrieve a snapshot summary
+
+Get details of a specific snapshot.
+
+```sh
+curl -s --url "https://api.aiven.io/v1/project/{project_name}/service/{service_name}/opensearch/_snapshot/aws-repo/first-snapshot" \
+--header "Authorization: Bearer $TOKEN"
+--header "Content-Type: application/json"
+```
+
+Example response:
+
+```json
+{
+  "snapshots": [
+    {
+      "snapshot": "first-snapshot",
+      "state": "SUCCESS",
+      "indices": ["test"],
+      "uuid": "7cdWedW7RC6FMSktlZTCDw"
+    }
+  ]
+}
+```
+
+### Delete a snapshot
+
+Delete a snapshot from a repository.
+
+```sh
+curl -s -X DELETE \
+--url "https://api.aiven.io/v1/project/{project_name}/service/{service_name}/opensearch/_snapshot/aws-repo/first-snapshot" \
+--header "Authorization: Bearer $TOKEN"
+--header "Content-Type: application/json"
+```
+
+Example response:
+
+```json
+{
+  "acknowledged": true
+}
+```
+
+## Error handling
+
+If an error occurs (such as `429`, `500`, or `503`), the Aiven API response includes the
```

Comment:
This is no longer true, e.g The Aiven API returns OpenSearch errors as they are. is the only thing needed.